src/main/java/com/google/devtools/build/lib/sandbox/SymlinkedSandboxedSpawn.java

```diff
@@ -88,7 +88,11 @@ public void filterInputsAndDirsToCreate(
 
   @Override
   protected void copyFile(Path source, Path target) throws IOException {
-    target.createSymbolicLink(source);
+    if (source.isDirectory()) {
+      target.createDirectory();
```

Comment:
Wouldn't this cause the sandboxed action to observe an empty directory if the source directory is the sole input?

I would add two integration test cases in bazel_sandboxing_test_sh:

1. Genrule receiving only a source directory
2. Genrule receiving both a source directory and a file inside it

In both cases, the genrule should assert that the file inside the directory exists.